package.json

```diff
@@ -332,13 +332,15 @@
     "format": "prettier --write .",
     "pretest": "npm run compile && tsc -p ./client",
     "test": "node ./client/out/test/runTest.js",
+    "coverage": "c8 --clean --check-coverage --lines 75 npm run test",
```

Comment:
I dont have a strong opinion on which one we choose. If we already have jest as a dependency (looks like we do) then we can reuse what we have.

Reply:
I also vote jest if c8 doesn't provide more features we need.

Reply:
It looks like the extension test api uses [mocha](https://code.visualstudio.com/api/working-with-extensions/testing-extension). Jest-Worker is being pulled in transitively, but the dev dependencies also support the mocha theory:

`  "devDependencies": {
    "@types/mocha": "^10.0.1",
    "@types/node": "^18.11.18",
    "@typescript-eslint/eslint-plugin": "^5.48.1",
    "@typescript-eslint/parser": "^5.48.2",
    "c8": "^7.12.0",
    "chai": "^4.3.6",
    "esbuild": "^0.13.15",
    "eslint": "^8.31.0",
    "media-typer": "^1.1.0",
    "mocha": "^10.2.0",
    "path-browserify": "^1.0.1",
    "prettier": "^2.7.1",
    "ts-loader": "^9.4.2",
    "typescript": "^4.4.3",
    "webpack": "^5.75.0",
    "webpack-cli": "^5.0.1"
  }`

Based on the original stackoverflow post from the issue, it looks like when using mocha that an additional library is needed.

Reply:
Another interesting data point are to look at use [trends](https://npmtrends.com/c8-vs-istanbul-vs-nyc ) for the 3 popular ones, nyc, c8, and istanbul:

Specifically with respect to downloads and the frequency for which the libraries are being updated.

Reply:
@2TomLi asked me to quick take a look at this since I've done a good amount of work on jest in other projects. I'm not familiar with this project though so if I miss something, sorry.

As @smorrisj was saying, this project, as best as I can see, is using mocha and I don't see any support for coverage directly from mocha like we get with jest. Using either c8 or nyc seems like an easy setup. istanbul doc says nyc is basically the istanbul CLI so you're left with just c8 or nyc. Both c8 and nyc seem popular enough to use and find support if needed. Both integrate the same way. So if y'all decide one isn't good, it should be an easy swap to the other.